fix: rpm fails if mirror tooling image not present

robert-cronin opened the pull request:
<!--
Please include the type of change in the PR title as "<type>: <summary>"
Valid <type> include: build, ci, docs, feat, fix, perf, refactor, test
See https://github.com/project-copacetic/copacetic/blob/master/CONTRIBUTING.md#pull-requests) for guidelines.
-->

Add fallback mechanism for when tooling image is not present for rpm-based images

Closes #976

robert-cronin commented:
I believe the specific image (a 1.0 tag from cbl-mariner/core) in question is not one that is expected to be present in the tooling images cache, and if so, it probably not an image we will see pulled often as it is fairly dated.

sozercan commented:
@robert-cronin do we want to add core:1.0 to the ghcr cache too?

sozercan reviewed (approved):
thank you! lgtm. added a few suggestions that we can handle as follow ups

robert-cronin commented:
> thank you! lgtm. added a few suggestions that we can handle as follow ups

Excellent thank you for your review! 

> @robert-cronin do we want to add core:1.0 to the ghcr cache too?
> 
> do we want to add the same mechanism for other package managers in case there's a new version but we are not caching it yet?

I've added the following based on suggestions above:

- https://github.com/project-copacetic/copacetic/issues/980
- https://github.com/project-copacetic/copacetic/pull/979

Also, I just noticed there was a call to `toolingImage.Validate` i forgot to clean up so that is latest push you'll notice in this PR
